furiosa-ai/device-api
# Refactor APIs with better error handling

Here are the main changes:
  * Downgrade tokio and update the cargo
  * Move tests to test_data to follow Rust directory hierarchy
  * Rename Core to Device
  * Rename CoreType to DeviceMode
  * Rename CoreStatus to DeviceStatus
  * Add more utility methods like fusible, available
  * Renames with_status to change_status
  * Renames DeviceStatus::Occupied2 to Fused
  * Adopt DeviceError to represent more specific errors
  * Improve the unit tests to make sure the API correctness
  * Remove async suffix from APIs
  * Rename list_async_with_path() to list_devices_with()
  * Rename collect_cores to reconcile_devices
  * Rename get_core to recognize_devices
  * Rename build_device_map_async to find_dev_files
  * Rename identify_device_type to identify_arch
